pkg/build/cmd/publishgithub.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/oauth2"
+)
+
+type repo struct {
+	owner string
+	name  string
+}
+
+type flags struct {
+	create       bool
+	dryRun       bool
+	tag          string
+	repo         *repo
+	artifactPath string
+}
+
+var (
+	errTokenIsEmpty    = errors.New("the environment variable GH_TOKEN must be set")
+	errTagIsEmpty      = errors.New(`failed to retrieve release tag from metadata, use "--tag" to set it manually`)
+	errReleaseNotFound = errors.New(`release not found, use "--create" to create the release`)
+)
+
+func PublishGitHub(ctx *cli.Context) error {
+	gitCtx := context.Background()
+	token := os.Getenv("GH_TOKEN")
+	f, err := getFlags(ctx)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	if f.tag == "" {
+		return cli.NewExitError(errTagIsEmpty, 1)
+	}
+
+	if token == "" {
+		return cli.NewExitError(errTokenIsEmpty, 1)
+	}
+
+	if f.dryRun {
+		return runDryRun(f, token, gitCtx)
+	}
+
+	client := newClient(gitCtx, token)
+	release, res, err := client.Repositories.GetReleaseByTag(gitCtx, f.repo.owner, f.repo.name, f.tag)
+	if err != nil && res.StatusCode != 404 {
+		return cli.NewExitError(err, 1)
+	}
+
+	if release == nil {
+		if f.create {
+			release, _, err = client.Repositories.CreateRelease(gitCtx, f.repo.owner, f.repo.name, &github.RepositoryRelease{TagName: &f.tag})
```

Comment:
@dsotirakis Yes, the flag `--create` would create the relase and add the assets, this would be useful since that's the current steps we take on our partner's mirrors.